### PR TITLE
make: Add "GOPATH not set"

### PR DESCRIPTION
**Pull request by teawater**
Most of the projects, they can be built with "make".  After that,
"sudo make install" can install the application.
It is not work for kata-runtime because kata-runtime's make must work
with golang in the environment that default sudo cannot supply it.
But "make install" doesn't need golang.

So add "GOPATH not set" to handle the issue.

Fixes: #1008

Signed-off-by: Hui Zhu <teawater@hyper.sh>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -14,24 +14,30 @@ done)
...
 ifeq ($(SKIP_GO_VERSION_CHECK),)
     include golang.mk
 endif
 
-ifeq ($(ARCH),)
-    ARCH = $(GOARCH)
-endif
+ifneq ($(GOPATH),)
+    ifeq ($(ARCH),)
+        ARCH = $(GOARCH)
```
Comment (jcvenegas) [inline Makefile]: you may want to move GOARCH=$(shell go env GOARCH) here

File: Makefile [hunk 2]
```diff
@@ -14,24 +14,30 @@ done)
...
+    endif
 
-ARCH_DIR = arch
-ARCH_FILE_SUFFIX = -options.mk
-ARCH_FILE = $(ARCH_DIR)/$(ARCH)$(ARCH_FILE_SUFFIX)
-ARCH_FILES = $(wildcard arch/*$(ARCH_FILE_SUFFIX))
-ALL_ARCHES = $(patsubst $(ARCH_DIR)/%$(ARCH_FILE_SUFFIX),%,$(ARCH_FILES))
+    ARCH_DIR = arch
```
Comment (jcvenegas) [inline Makefile]: none of the following variables are dependent of GOPATH

File: Makefile [hunk 3]
```diff
@@ -498,6 +505,10 @@ show-summary: show-header
 	@printf "• golang:\n"
 	@printf "\t"
 	@go version
+else
+	@printf "• No golang mode:\n"
```
Comment (grahamwhaley) [inline Makefile]: suggest a reword here:
```
    @printf "• GOPATH not set:\n"
    @printf "\tCan only install prebuilt binaries\n"
```

File: Makefile (hunk 2 above)
Comment (teawater) [reply Makefile]: If they are moved out from "ifneq ($(GOPATH),)" will get following error without GOPATH:
sudo make install
[sudo] password for teawater:
Makefile:39: arch/-options.mk: No such file or directory
make: *** No rule to make target 'arch/-options.mk'.  Stop.

**Comment by lifupan** [comment]
LGTM

**Comment by gnawux** [comment]
LGTM

Actually, I found this issue during the kubecon demo rehearsal, and thanks @teawater for solving it.

> @jcvenegas 
> but agree probably wont need to block if non of the GOLANG variables are not set.

Yes, this is my point.

**Pull request merged**